rust-runtime/aws-smithy-http-tower/src/map_request.rs

```diff
@@ -97,10 +97,15 @@ where
     }
 }
 
-#[pin_project(project = EnumProj)]
-pub enum MapRequestFuture<F, E> {
-    Inner(#[pin] F),
-    Ready(Option<E>),
+pin_project! {
+    #[project = EnumProj]
+    pub enum MapRequestFuture<F, E> {
+        Inner {
```

Comment:
Not really related to your changes, but the name `Inner` is kind of confusing to me here. Maybe we should have called it `Pending` or something? Is there a broader idiomatic pattern for this with futures?

Reply:
Looking at it, I would name the variants `Pending` and `Complete`. @rcoh what do you think?